shipyard_blueprints/bigquery/shipyard_bigquery/utils/exceptions.py

```diff
@@ -0,0 +1,50 @@
+from shipyard_templates import ExitCodeException
+
+
+class DatasetNotFound(ExitCodeException):
+    def __init__(self, message: str, exit_code: int):
+        super().__init__(message, exit_code)
+        self.message = message
+        self.exit_code = exit_code
```

Comment:
I would suggest with the custom exceptions to include the exit code and possibly the message as well.